src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/Sample.java

```diff
@@ -42,6 +44,11 @@
     @JsonProperty
     private String name;
 
+    @ApiModelProperty(position = 3, dataType = "java.lang.String", example = "[url1, url2]", notes = "Url to Taxonomy List")
+    @JsonProperty
+    @ManyToMany
+    private List<Taxonomy> taxonomies;
```

Comment:
I guess we need to validate that a sample should contain at-least one Taxonomy.

Reply:
To my knowledge in manytomany relation right side can be 0,1 or more. The very @ManyToMany does not provide `optional`. I explored other JPA annotations including @Check, @Joinxxx; nothing works better.

I tried enforcing in code using @PrePersist and @PreUpdate. In prepersist, taxonomy URL list will always be null. To fix this we need to use content-type as `uri-list`. But it can be used only after creating Sample object with Json body and then updating with Taxonomy uri-list. After creating 
 @PrePersist will not be calledagain; @PreUpdate will get called. 

With this we need to enforce customer to create Sample without taxonomy and then update with taxonomy list. I need to explore this more to make sure it works perfectly as expected. The other classical solution is to override all calls like put,post... without utilizing framework benefits.

Reply:
the taxonomy being null during post because of Mockmvc only , please try to look for possible alternate if any, and also you can try with custom validators instead of PrePersist and updates as well .

Reply:
Yes, i am exploring @NotNull and other annotations in parallel.

Reply:
Done

Reply:
@selvaebi : It just checks whether the list is null or not i guess, not the elements in it .
@iyangar :  I and selva observed T2D-218 changes helps to resolve this issue. Details in Jira

Reply:
Done